core/controllers/acl_decorators.py

```diff
@@ -48,12 +48,23 @@
 from core.domain import topic_services
 from core.domain import user_services
 
-from typing import Any, Callable, Dict, List, Optional, Type, TypeVar
+from typing import (TYPE_CHECKING, Any, Callable, Dict, List, Optional, Type,
+                    TypeVar)
 
 # Note: '_SelfBaseHandlerType' is a private type variable because it is only
 # supposed to denote the 'self' argument of the handler function that the
 # decorator is decorating. So, do not make it public type variable in future.
-_SelfBaseHandlerType = Type[base.BaseHandler]
+
+if TYPE_CHECKING:
+    # Here we use type Any because type parameters for generic type
+    # "BaseHandler" is missing, hence used Any to type annotate "BaseHandler"
+    # class.
+    BaseHandler = base.BaseHandler[Any, Any]
```

Comment:
@seanlip, Here I added a comment for `Any` type but seems like lint checks are not recognizing it and still giving below error message.
Do you know where should I fix it?
```
Any type is used. If the Any type is really needed, then please add a proper comment with clear justification why other specific types cannot be used. The format of the comment should be -> Here we use type Any because ...
```

Reply:
I think you can find it by searching the codebase for "If the Any type".

My guess is that the lint check is not checking for this sort of construct. Is it possible to keep something that's more similar to the previous logic here?

Reply:
> Is it possible to keep something that's more similar to the previous logic here?

@seanlip I am not sure what you meant by this. Can you please explain?

Reply:
I don't see anything that looks like `BaseHandler = base.BaseHandler[Any, Any]` in the current file, so I was wondering why we need to add it and why we can't use the existing approach instead.

Reply:
https://stackoverflow.com/questions/79397624/missing-type-parameters-for-generic-type-strictredis

Reply:
OK, so keeping the existing approach won't work / results in errors? I still don't really understand what was wrong with Type[base.BaseHandler].

Reply:
With this line
```
_SelfBaseHandlerType = Type[base.BaseHandler]
```
Mypy is complaining `Missing type parameters for generic type "BaseHandler"  [type-arg]`

Reply:
For now, I will choose to add ignore, any thoughts?

Reply:
We can create an issue to track this later

Reply:
Actually, looking at this again, what is the BaseHandler on the left? Is it a type or is it a class (that's equivalent to base.BaseHandler)?

If the former we should name it more differently from BaseHandler. I'm not really familiar with this syntax...

Reply:
type
![image](https://github.com/user-attachments/assets/d026cc15-f3a0-4df4-ad01-91d38c5bd1f8)